app/uk/gov/hmrc/hec/controllers/DESController.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.controllers
+
+import cats.data.Validated._
+import cats.implicits._
+import com.google.inject.{Inject, Singleton}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import uk.gov.hmrc.hec.models.ids.CRN
+import uk.gov.hmrc.hec.services.DESService
+import uk.gov.hmrc.hec.services.DESService.{BackendError, DESError, DataNotFoundError, InvalidCRNError}
+import uk.gov.hmrc.hec.util.Logging
+import uk.gov.hmrc.hec.util.Logging.LoggerOps
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class DESController @Inject() (
+  DESService: DESService,
+  cc: ControllerComponents
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc)
+    with Logging {
+
+  private def handleError(e: DESError) = e match {
+    case DataNotFoundError(msg) =>
+      logger.warn(msg)
+      NotFound
+    case InvalidCRNError(msg)   =>
+      logger.warn(s"Invalid CRN - $msg")
+      BadRequest("Invalid CRN")
```

Comment:
Should this error message match with the one we have in api doc? if no, then it's ok

Reply:
don't think so, can't see why we have match it